code/io/timer.cpp

```diff
@@ -104,7 +106,12 @@ std::uint64_t timer_get_nanoseconds()
 {
 	auto time = get_performance_counter();
 
-	return (time * NANOSECONDS_PER_SECOND) / Timer_perf_counter_freq;
+	if (Timer_perf_counter_freq < NANOSECONDS_PER_SECOND) {
+		// This can cause overflows but I don't know a better solution...
+		return (time * NANOSECONDS_PER_SECOND) / Timer_perf_counter_freq;
+	} else {
+		return time / (Timer_perf_counter_freq / NANOSECONDS_PER_SECOND);
```

Comment:
This will result in a loss of precision, are you OK with that?

Example: imagine if the freq is 1.8 GHz like an XBone, the time returned will be time / (1.8/1.0) -> time /1.0 due to long division.

Also, can dividing by the GCD be useful?

Finally, the divided value can probably be cached instead of recalculated every time.